### PR TITLE
Added `ServiceMethod` and `ReturnType`.

### PR DESCRIPTION
**Pull request by vcolin7**

### CODE REVIEW

File: sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/annotation/ReturnType.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.annotation;
+
+/**
+ * Enumeration of return types used with {@link ServiceMethod} annotation to indicate if a method is expected to return
+ * a single item or a collection.
+ */
+public enum ReturnType {
```
Comment (samvaity) [inline sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/annotation/ReturnType.java:10]: do we not need any rest proxy impl c changes o handle the return type?

Comment (vcolin7) [reply sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/annotation/ReturnType.java:10]: From having a look at the Azure Core counterpart, no. It seems this is only for static code analysis tools like CheckStyle and for when we use APIView.

Comment (vcolin7) [reply sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/annotation/ReturnType.java:10]: Maybe @JonathanGiles or @srnagar have a different point of view.

**Review by alzimmermsft — APPROVED** [review]

File: sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/annotation/ReturnType.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    /**
+     * Simple collection, enumeration, return type.
+     */
+    COLLECTION,
+
+    /**
+     * Long-running operation return type.
+     */
+    LONG_RUNNING_OPERATION
```
Comment (srnagar) [inline sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/annotation/ReturnType.java:24]: We don't have long running operations in unbranded libraries. So, this should be removed.

**Pull request merged**